libraries/AP_Mission/AP_Mission_Relative.cpp

```diff
@@ -0,0 +1,311 @@
+        _translation.calculated = false;
+        _translation.do_translation = true;
+
+        // memorize the Basepoint (location of switching to AUTO)
+        AP::ahrs().get_position(_basepoint_loc); // _basepoint_loc.alt is absolute in [cm] in every case
+
+        AP_Mission::Mission_Command tmp;
+
+        // check if one of the Waypoints has (terrain_alt == 1) -> no translation allowed
+        uint16_t i=0;
+        while (AP::mission()->get_next_nav_cmd(i, tmp) && (_translation.do_translation) && (tmp.id != MAV_CMD_DO_LAND_START)) { // advance to the next command
+            // check if Waypoint with Location
+            if (!(tmp.content.location.lat == 0 && tmp.content.location.lng == 0)) {
+                switch (tmp.id) { // no translation for TAKEOFF or LAND commands
+                    case MAV_CMD_NAV_TAKEOFF:
+                    case MAV_CMD_NAV_LAND:
+                    case MAV_CMD_NAV_VTOL_TAKEOFF:
+                    case MAV_CMD_NAV_VTOL_LAND:
+                        break;
+                    default:
+                        if (tmp.content.location.terrain_alt == 1) {
+                            _translation.do_translation = false;
+                            gcs().send_text(MAV_SEVERITY_NOTICE, "MR: terrain_alt -> NO MOVE POSSIBLE");
+                            return;
+                        }
+                }
+            }
+            i++;// move on to next command
+        }
+
+        // no translation if the distance to Homepoint is too small
+        tmp.content.location = AP::ahrs().get_home();
+        float tmp_distance = tmp.content.location.get_distance(_basepoint_loc); // in [m]
+        if (tmp_distance < _no_translation_radius) {
+             gcs().send_text(MAV_SEVERITY_NOTICE, "MR: distance to Home: %.1fm -> NO translation", tmp_distance);
+            _translation.do_translation = false;
+            return;
+        }
+
+        // calculate amount of rotation
+        switch (_restart_behaviour) {
+            case Restart_Behaviour::RESTART_ROTATED_NORTH:
+            case Restart_Behaviour::RESTART_ROTATED_FIRST_WP:
+                tmp.content.location = AP::ahrs().get_home();
+                // get direction from Homepoint to Basepoint (North-related)
+                _translation.direction = tmp.content.location.get_bearing_to(_basepoint_loc);   // in centidegrees from 0 36000
+                break;
+            case Restart_Behaviour::RESTART_ROTATED_HEADING:
+                // rotation via heading at Basepoint
+                _translation.direction = AP::ahrs().yaw_sensor;
+                break;
+            case Restart_Behaviour::RESTART_ROTATED_CHANNEL:
+                // rotation via RC-channel: 1000mys<->-180deg 1500mys<->0deg 2000mys<->+180deg
+                uint16_t radio_in;
+                if (_rotate_ch > 0) {
+                    radio_in = RC_Channels::get_radio_in(_rotate_ch-1);
+                    radio_in = (radio_in > 2000 ? 2000 : radio_in);
```

Comment:
I think there are global defines for these two values.

Reply:
I renamed radio_in, but what's the second value you mentioned?

Reply:
One was the 1000, the second was the 2000.
I think we already have fuctions that return the value of a switch (basicaly the function cheks if the pwm value is bellow 1000, or above 2000) So there is no need to have the 1000 and 2000 values hardcoded here.

Reply:
It's not a switch, it's a standardized value for rotation:
1000µs = -180deg ... 1500µs = 0deg ... 2000µs = +180deg
(I know, docu is missing - When that draft-PR passes all checks, I will attach the draft-documentation)

Reply:
Is it saving memory if I would define it as static const uint16_t ?
What do you recommend?